### PR TITLE
Use remote repository instead of local jar files | Remove upnplib

### PR DESCRIPTION
**Pull request by RoiEXLab**
This removes the jar files located at ./lib
(Veqryn actually compiled those files using ant on his local machine according to the MANIFEST file of upnp-lib ^^)
I just needed to change a single line of code to match changes with param order.
Because UPNP on routers is considered unsafe and therefore almost no routers have enabled it by default (because it well actually enables EVERY device/application on your network to expose your local network to the web). I'd like to remove the usage of this lib altogether.

Thoughts? @DanVanAtta @ron-murhammer @ssoloff

### CODE REVIEW

**Comment by DanVanAtta** [comment]
Build:
```
Execution failed for task ':checkstyleMain'.
> Checkstyle rule violations were found. See the report at: file:///home/travis/build/triplea-game/triplea/build/reports/checkstyle/main.html
```

Otherwise LGTM @RoiEXLab 

AFAIK the auto-configure of firewall generally does not work. It would make sense, a firewall where any client can open up rules, is not much of a firewall. From my perspective, it would be okay to remove UPNP altogether.

**Comment by RoiEXLab** [comment]
@DanVanAtta Should I just remove it in this PR?

**Comment by RoiEXLab** [comment]
@DanVanAtta I removed the usage of upnplib!

**Comment by DanVanAtta** [comment]
:+1:  Thanks.

Leaving open for a bit in case @ron-murhammer  or others have additional insights why removing UPNP would be bad.

**Comment by ron-murhammer** [comment]
@RoiEXLab @DanVanAtta Won't this lead to even fewer lobby users being able to host? I don't particularly have a strong opinion on this as I don't play lobby games but given the limited number of bots and limitations in bot hosted games, I would think this still adds value.

**Comment by RoiEXLab** [comment]
@ron-murhammer All this library does for us is adding a port forward rule to the default gateway.
Given that this is a security issue for routers as this doesn't even require a password, most routers don't support this anyways.
It's really not that difficult doing this manually, but it differs from router model to router model.
TL;DR if people really want to host their own games they will also be able to spend a couple minutes setting up the rule manually

**Comment by ron-murhammer** [comment]
@RoiEXLab Understood. Though I think you overestimate people's technical ability. I'm not sure what percentage of the time this works but I know there are a lot of players that won't even consider or have any idea how to configure their router.

**Comment by RoiEXLab** [comment]
Portforwarding is a common practice when it comes to game servers. Try googling Portforwarding <your router model> and you will at least get 3 results. 
Probably not the best quality but sufficient.

**Comment by ron-murhammer** [comment]
@RoiEXLab @DanVanAtta Agree but that still doesn't mean most people can figure it out. I would suggest posting a thread of the forum to see if any of the mods have strong opinions on this. If they don't see to care about it and @DanVanAtta wants to remove it then I'm fine with it.

**Comment by DanVanAtta** [comment]
FWIW I've never heard of any examples of this feature working, IMO it even gets in the way as an extra step. Assigning to @RoiEXLab to check with wider community on usage.

**Comment by RoiEXLab** [comment]
https://forums.triplea-game.org/topic/135
The Polls are open

**Comment by RoiEXLab** [comment]
@DanVanAtta @ron-murhammer It seems @prastle is able to use UPnP to do portforwarding on his router...
UPnP is considered insecure, primarily because (standard) UPnP supports no authentification (there seem to be custom extensions that "fix" this issue, but no commonly adapted standards). (Also some exploits were found which made routers vulnerable for request from outside of the local network)
Found a couple sites regarding this Topic:
 - https://security.stackexchange.com/a/118951
 - https://www.howtogeek.com/122487
 - (Also a couple german articles ^^)

TL;DR UPnP comes with security risks and therefore common advice is to disable UPnP completely.

I believe we shouldn't encourage users to enable it just for easier configuration...

**Comment by RoiEXLab** [comment]
@ron-murhammer Bumping...

**Comment by ron-murhammer** [comment]
@RoiEXLab I'm fine with this as long as you and @DanVanAtta are prepared to handle any potential backlash from lobby users that were using it.

@DanVanAtta I'll leave this one for you to take a final pass over and merge.

File: build.gradle [hunk 1]
```diff
@@ -68,9 +68,7 @@ dependencies {
     compile 'org.mindrot:jbcrypt:0.3m'
     compile 'org.yaml:snakeyaml:1.17'
     compile 'org.json:json:20160810'
-
-    compile files('lib/AppleJavaExtensions-1.6.jar')
```
Comment (DanVanAtta) [inline build.gradle]: One question - AppleJavaExtensions was being used for UPNP? Or is it just unused?

Comment (RoiEXLab) [inline build.gradle]: it's called orange-extensions in maven central
